index.rst

```diff
@@ -2,28 +2,99 @@
+This service has been referred  to as Catcher.
+In essence, the Catcher is not that different from the Watcher or other stream processing off-the-shelf technologies like Kapacitor.
+This technote explores the idea of using influx2.0 new stream processing capabilities to as an alternative implementation for the Catcher.
 
-.. note::
+Introduction
+============
 
-   **This technote is a work-in-progress.**
+One of the finding of the First-Look Analysis and Feedback Functionality (FAFF) working group :sitcomtn:`025` :cite:`SITCOMTN-025`, was the need for a service to ...
 
-Abstract
-========
+   handle the low-level coordiation of identifying when a specific condition is met, then launching and monitoring an analysis process.
+
+The original proposal from the FAFF working group was to develop a Commandable SAL Component (CSC), similar to the Watcher, that would provide be capable of monitoring data from the system and trigger certain actions (e.g. perform some computation and/or generate reports) based on some pre-defined conditions.
+
+Following additional discussions, we have decided to experiment with an alternative approach, replacing the CSC with an off-the-shelf solution provided by the Engineering Facility Database (EFD).
+
+As shown in :sqr:`034` :cite:`SQR-034`, the EFD has of an `influxDB <https://www.influxdata.com/>`__ time series database, that stores the data produced by the system.
+The release of influx2.0, brings a number of important updates, amongst which two are relevant for stream processing; `flux`_ and `influxDB task`_.
+
+.. _flux: https://docs.influxdata.com/flux/v0.x/
+.. _influxDB task: https://docs.influxdata.com/influxdb/cloud/process-data/get-started/
+
+
+`flux`_ is a new functional data scripting language that is designed to process data streams.
+It contains extensive libraries that allows users to express complex operations, compatible with what can be done with programming languanges.
+Furthermore, `influxDB task`_ provides a way to schedule `flux`_ scripts.
+
+
+Each Catcher "rule" or "alarm" would be comprised of an `influxDB task`_ to execute a user defined `flux`_ script periodically.
+The `flux`_ script performs a set of actions/calculations based on EFD data, additional processes/tasks can be triggered depending on the results of these calculations and results can be stored in the EFD for future reference.
+
+Below we describe how some of the Catcher functionality would be implemented.
+
+Alarms
+------
+
+One of the key features expected from the Catcher was the ability to issue alarms that can be displayed in LOVE.
+One option to support this feature would be to create a custom LOVE producer that can be interacted with using a REST API.
+The alert information would be persisted in the EFD by the `flux`_ script, which would then post a message to the producer of the new alert.
+The producer would then read the alert informatino from the EFD and display it.
```

Comment:
.. information ...

Does LOVE provide a REST API? if so, I think a better implementation of this would be to send a POST request to an `alert` endpoint on that API, instead of polling the EFD for new messages.

Reply:
So, we will have to develop the backend to support this so, in principle, it can have whatever we think is best. 

That being said, I was thinking of using the REST API only to tell LOVE that there is a new `alert` it needs to look for. 

The problem is persisting information. LOVE doesn't have a local storage. So any information you need to send to LOVE needs to either be stored in the "producer", in the EFD or some other DB LOVE can access.. If we use the REST API to send information about the `alert`, LOVE's backend would have to store is somewhere or it would be lost between restarts. Since we should probably have this in the EFD anyway, I though relying on EFD to get the information from would be the simplest solution.